src/storage/azure/storage.c

```diff
@@ -0,0 +1,729 @@
+    };
+
+    if (data->level >= storageInfoLevelBasic)
+    {
+        info.type = type;
+        info.size = type == storageTypeFile ?
+            cvtZToUInt64(strPtr(xmlNodeContent(xmlNodeChild(xml, AZURE_XML_TAG_CONTENT_LENGTH_STR, true)))) : 0;
+        info.timeModified = type == storageTypeFile ?
+            httpDateToTime(xmlNodeContent(xmlNodeChild(xml, AZURE_XML_TAG_LAST_MODIFIED_STR, true))) : 0;
+    }
+
+    data->callback(data->callbackData, &info);
+
+    FUNCTION_TEST_RETURN_VOID();
+}
+
+static bool
+storageAzureInfoList(
+    THIS_VOID, const String *path, StorageInfoLevel level, StorageInfoListCallback callback, void *callbackData,
+    StorageInterfaceInfoListParam param)
+{
+    THIS(StorageAzure);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(STORAGE_AZURE, this);
+        FUNCTION_LOG_PARAM(STRING, path);
+        FUNCTION_LOG_PARAM(ENUM, level);
+        FUNCTION_LOG_PARAM(FUNCTIONP, callback);
+        FUNCTION_LOG_PARAM_P(VOID, callbackData);
+        FUNCTION_LOG_PARAM(STRING, param.expression);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(path != NULL);
+    ASSERT(callback != NULL);
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        StorageAzureInfoListData data = {.level = level, .callback = callback, .callbackData = callbackData};
+        storageAzureListInternal(this, path, param.expression, false, storageAzureInfoListCallback, &data);
+    }
+    MEM_CONTEXT_TEMP_END();
+
+    FUNCTION_LOG_RETURN(BOOL, true);
+}
+
+/**********************************************************************************************************************************/
+static StorageRead *
+storageAzureNewRead(THIS_VOID, const String *file, bool ignoreMissing, StorageInterfaceNewReadParam param)
+{
+    THIS(StorageAzure);
+
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(STORAGE_AZURE, this);
+        FUNCTION_LOG_PARAM(STRING, file);
+        FUNCTION_LOG_PARAM(BOOL, ignoreMissing);
+        (void)param;                                                // No parameters are used
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(file != NULL);
+
+    FUNCTION_LOG_RETURN(STORAGE_READ, storageReadAzureNew(this, file, ignoreMissing));
+}
+
+/**********************************************************************************************************************************/
+static StorageWrite *
+storageAzureNewWrite(THIS_VOID, const String *file, StorageInterfaceNewWriteParam param)
+{
+    THIS(StorageAzure);
+
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(STORAGE_AZURE, this);
+        FUNCTION_LOG_PARAM(STRING, file);
+        (void)param;                                                // No parameters are used
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(file != NULL);
+    ASSERT(param.createPath);
+    ASSERT(param.user == NULL);
+    ASSERT(param.group == NULL);
+    ASSERT(param.timeModified == 0);
```

Comment:
Yes. Just a sanity check based on the driver capabilities.

Reply:
OK, just seemed weird given : 
`(void)param;                                                // No parameters are used`

Reply:
I'd like to move these asserts into the storage interface anyway. param.createPath could go now since we have a feature for paths but the others would need features, e.g. `storageFeatureUserGroupWrite`, `storageFeatureTimeWrite`. Or something like that. Then the storage interface could check that the driver supports the parameters and we could simplify the driver a bit.